pkg/reduce/readloop/readloop.go

```diff
@@ -94,6 +95,7 @@ func NewReadLoop(ctx context.Context,
 		whereToDecider:        whereToDecider,
 		publishWatermark:      pw,
 		udfInvocationTracking: make(map[partition.ID]*task),
+		latestWatermark:       processor.Watermark(time.Time{}),
```

Comment:
rename this to indicate that you are tracking the window boundary

Reply:
I am not tracking the window boundary, I am tracking the latest watermark we have received